doc/development/plugins.rst

```diff
@@ -232,6 +232,91 @@ or can include in-built transforms such as:
 * :func:`pennylane.devices.preprocess.validate_multiprocessing_workers`
 * :func:`pennylane.devices.preprocess.validate_adjoint_trainable_params`
 * :func:`pennylane.devices.preprocess.no_sampling`
+  
+**Decompositions:**
+
+The :func:`~pennylane.devices.preprocess.decompose` transform is typically required as part of the transform program that
+decomposes unsupported operations to the device's native gate set. To define this transform a stopping condition needs
+to be specified. This is a function mapping an operator to a boolean that determines whether the operator should be decomposed.
+
+For example, for a device supporting the ``CNOT`` gate, the stopping condition and the decompose transform
+can be specified like so:
+
+>>> from pennylane.devices.preprocess import decompose
+>>> def stopping_condition(op):
+...     return op.name == "CNOT"
+>>> tape = qml.tape.QuantumScript([qml.CNOT(wires=(0,1))], [qml.expval(qml.Z(0))])
+>>> batch, fn = qml.devices.preprocess.decompose(tape, stopping_condition)
+>>> batch[0].circuit
+[CNOT(wires=[0, 1]), expval(Z(0))]
+
+However, if the device native gate set is unreachable with the default decompositions defined in PennyLane,
+an error will be raised. In this case, you may need to override the decompositions of certain operators
+via the ``decomposer`` argument. 
+
+For example, consider we have a device with ``RX``, ``RY`` and ``IsingXX`` as native gates but we want
+to execute a circuit written in terms of ``CNOT`` s. Then, we can define a decomposition for ``CNOT`` 
+(e.g., ``custom_decomposer``) and pass it to the decomposer:
+
+.. code-block:: python
+
+    def stopping_condition(op):
+        return op.name in {"IsingXX", "RX", "RY"}
+
+    def custom_decomposer(op):
+        if isinstance(op, qml.CNOT):
+            wires = op.wires
+            return [
+                qml.RY(np.pi/2, wires=wires[0]),
+                qml.IsingXX(np.pi/2, wires=wires),
+                qml.RX(-np.pi/2, wires=wires[0]),
+                qml.RY(-np.pi/2, wires=wires[0]),
+                qml.RY(-np.pi/2, wires=wires[1])
+            ]
+        return op.decomposition()
+
+>>> circuit = qml.tape.QuantumScript([qml.CNOT(wires=(0,1))], [qml.expval(qml.Z(0))])
+>>> batch, fn = decompose(circuit, stopping_condition, decomposer=custom_decomposer)
+>>> batch[0].circuit
+[RY(1.5707963267948966, wires=[0]),
+IsingXX(1.5707963267948966, wires=[0, 1]),
+RX(-1.5707963267948966, wires=[0]),
+RY(-1.5707963267948966, wires=[0]),
+RY(-1.5707963267948966, wires=[1]),
+expval(Z(0))]
+
+An alternative approach for overriding the decompositions of certain operators is by taking advantage of
+the new graph-based decomposition algorithm.
```

Comment:
```suggestion
the new graph-based decomposition algorithm, with :func:`<qml.decomposition.enable_graph()> pennylane.decomposition.enable_graph`.
```

Reply:
This func link was broken so I tried: 
:func:`qml.decomposition.enable_graph() <~pennylane.decomposition.enable_graph>` which doesn't work either...

An alternative is using: 
:func:`~.decomposition.enable_graph()`
but it renders just as `enable_graph()`